### PR TITLE
UnitType property added to RevitParameter and RevitParameterDifference

### LINKED ISSUE #1315: Introduce unit type information to RevitParameter and RevitParameterDifference
#### Description:

In diffing workflows it is not always obvious what unit type does a given parameter (or its difference) represent. Therefore, it would be really helpful to pass this information from parameter definition.

### PR DESCRIPTION
**Pull request by pawelbaran**
### Issues addressed by this PR

Closes #1315

### Test files

On [SharePoint](https://burohappold.sharepoint.com/sites/BHoM/02_Current/Forms/AllItems.aspx?RootFolder=%2Fsites%2FBHoM%2F02%5FCurrent%2F12%5FScripts%2F02%5FPull%20Request%2FBHoM%2FRevit%5FToolkit%2F%231316%2DUnitTypeOnParam&FolderCTID=0x0120008122C8891F89054B8ACED0196C70DFC4) - run the script on chosen Revit file/version.

### CODE REVIEW

**Comment by pawelbaran** [comment]
@BHoMBot check core
@BHoMBot check compliance

**Comment by pawelbaran** [comment]
@BHoMBot check versioning
@BHoMBot check installer

**Comment by pawelbaran** [comment]
@BHoMBot check versioning
@BHoMBot check installer

**Comment by pawelbaran** [comment]
@BHoMBot check versioning

**Comment by pawelbaran** [comment]
@BHoMBot check versioning

File: Revit_Core_Engine/Query/UnitTypeByPropertyName.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            m_UnitTypesWithIdentifiers = new Dictionary<string, ForgeTypeId>();
+            foreach (PropertyInfo info in typeof(SpecTypeId).GetProperties())
+            {
+                ForgeTypeId unitType = info.GetValue(null) as ForgeTypeId;
+                if (unitType != null)
+                    m_UnitTypesWithIdentifiers.Add(info.Name, unitType);
+            }
+        }
+
+        private static Dictionary<string, ForgeTypeId> m_UnitTypesWithIdentifiers = null;
```
[the remark this answers is not in the document]
Comment (pawelbaran) [reply Revit_Core_Engine/Query/UnitTypeByPropertyName.cs:107]: I do not mind if that's the only complaint about the code 😃 Fixed!

**Comment by pawelbaran** [comment]
@BHoMBot check compliance
@BHoMBot check core

**Comment by pawelbaran** [comment]
@BHoMBot check compliance
@BHoMBot check core

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance
@BHoMBot check core

**Comment by FraserGreenroyd** [comment]
@BHoMBot check null-handling
@BHoMBot check serialisation
@BHoMBot check versioning

**Comment by pawelbaran** [comment]
@BHoMBot check versioning

**Comment by pawelbaran** [comment]
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check installer

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am requesting neutral checks on: versioning

**Pull request merged**